_shared/local_extensions.py

```diff
@@ -219,6 +220,29 @@ def has_services(self, context):
             context, "docker-compose/services.yml"
         )
 
+    @pass_context
+    def pytest_plugins(self, context, tests_type):
+        """Return the list of pytest plugins that should be activated for this project."""
+        cookiecutter = context["cookiecutter"]
+
+        plugins = ["tests.pytest_plugins.factory_boy"]
+        directory = cookiecutter["_directory"]
+        postgres = cookiecutter.get("postgres") == "yes"
+        is_app = directory in ["pyapp", "pyramid-app"]
+
+        if directory == "pyramid-app":
+            plugins.append("tests.pytest_plugins.pyramid")
+        if is_app and postgres:
+            plugins.append("tests.pytest_plugins.db")
+        if tests_type == "unit" and is_app and postgres:
+            plugins.append("tests.unit.pytest_plugins.db")
+        if tests_type == "unit" and directory == "pyramid-app":
+            plugins.append("tests.unit.pytest_plugins.pyramid")
+        if tests_type == "functional" and is_app and postgres:
+            plugins.append("tests.functional.pytest_plugins.db")
+
+        return plugins
+
```

Comment:
Different pytest plugins are enabled depending on various factors, I encapsulated the logic in this function.